src/constructors/constructWithOptions.js

```diff
@@ -11,6 +11,8 @@ export default (css: Function) => {
     }
 
     /* This is callable directly as a template function */
+    // TODO - try modifying this to automatically call extends
+    // and remove all extra wrapper components
```

Comment:
Oh I see, I wasn't even thinking about w.r.t. `template` and its functions being dynamically created vs living off one prototype and being optimized?

I was more-so thinking about wrapper calls to `styled(AnotherStyledComponent)` (and `.extend`). What do you think of creating a single component with merged rules as opposed to wrapping in a higher-order styled component. I'm fairly new to this code, please feel free to point misunderstandings with the current approach :). 

I dislike that this would create extra classnames. It seems like the most idiomatic way to use `styled-components` is to create numerous primitives, and then `style(Primitive)` as needed. In the codebase I use we end up with a lot wrapping components because of this.

(There's also a tradeoff here w.r.t. this optimization potentially decreasing the amount of times the `static` optimization in this patch actually can be used :) )